src/rust/engine/src/nodes.rs

```diff
@@ -812,6 +807,8 @@ impl Task {
           .expect("edges for task exist.")
           .entry_for(&select_key)
           .unwrap_or_else(|| {
+            // TODO: convert this to a Result, as well as all other panic!()s and .expects() that
+            // assert graph logic which is not clearly going to be correct in all future cases.
```

Comment:
While I agree in general, the fix here is probably to improve the `RuleGraph`'s enums and types to make these operations infalliable, rather than turning this into a user-visible error. A failure here would be a bug in the `RuleGraph`, rather than in a user's rules.

Reply:
This sounds good, will edit the comment to match.

Reply:
Turned into a comment about returning enums to avoid flattening failure modes, let me know if that captures the intent.